zone/client_packet.cpp

```diff
@@ -11794,578 +11853,658 @@ void Client::Handle_OP_RaidCommand(const EQApplicationPacket *app)
+									if (raid->IsLocked()) {
+										raid->SendRaidLockTo(c);
+									}
 								}
+								else {
+									Client* c = nullptr;
 
-								raid->SendRaidCreate(c);
-								raid->SendMakeLeaderPacketTo(raid->leadername, c);
-								if (group->IsLeader(group->members[x]))
-									raid->AddMember(c, free_group_id, false, true);
-								else
-									raid->AddMember(c, free_group_id);
-								raid->SendBulkRaid(c);
-								if (raid->IsLocked()) {
-									raid->SendRaidLockTo(c);
+									if (sending_invite_member->IsClient())
+										c = sending_invite_member->CastToClient();
+									else
+										continue;
+
+									raid->SendRaidCreate(c);
+									raid->AddMember(c, raid_free_group_id);
+									raid->SendBulkRaid(c);
+									raid->SendMakeLeaderPacketTo(raid->leadername, c);
+									if (raid->IsLocked()) {
+										raid->SendRaidLockTo(c);
+									}
 								}
 							}
 						}
-						group->JoinRaidXTarget(raid);
-						group->DisbandGroup(true);
-						raid->GroupUpdate(free_group_id);
+						player_sending_invite_group->JoinRaidXTarget(raid, true);
+						player_sending_invite_group->DisbandGroup(true);
+						raid->GroupUpdate(raid_free_group_id);
+						raid_free_group_id = raid->GetFreeGroup();
 					}
 					else {
-						raid->SendRaidCreate(this);
-						raid->SendMakeLeaderPacketTo(raid->leadername, this);
-						raid->AddMember(this);
-						raid->SendBulkRaid(this);
-						if (raid->IsLocked()) {
-							raid->SendRaidLockTo(this);
-						}
+						raid->SendRaidCreate(player_sending_invite);
+						raid->AddMember(player_sending_invite, RAID_GROUPLESS, true, false, true);
 					}
-				}
-				else
-				{
-					Group *player_invited_group = player_accepting_invite->GetGroup();
-					Group *group = GetGroup();
-					if (group) //if our target has a group
-					{
-						raid = new Raid(player_accepting_invite);
-						entity_list.AddRaid(raid);
-						raid->SetRaidDetails();
 
-						uint32 raid_free_group_id = raid->GetFreeGroup();
-
-						/* If we already have a group then cycle through adding us... */
-						if (player_invited_group) {
-							Client *client_to_be_leader = nullptr;
-							for (int x = 0; x < 6; x++) {
-								if (player_invited_group->members[x]) {
-									if (!client_to_be_leader) {
-										if (player_invited_group->members[x]->IsClient()) {
-											client_to_be_leader = player_invited_group->members[x]->CastToClient();
-											raid->SetGroupLeader(client_to_be_leader->GetName());
-										}
-									}
-									if (player_invited_group->IsLeader(player_invited_group->members[x])) {
-										Client *c = nullptr;
-
-										if (player_invited_group->members[x]->IsClient())
-											c = player_invited_group->members[x]->CastToClient();
-										else
-											continue;
+					Client* client_to_add = nullptr;
+					/* Add client to an existing group */
+					for (int x = 0; x < MAX_GROUP_MEMBERS; x++) {
```

Comment:
You can range through the array versus doing the integer iteration

eg

```
for (auto & m : members) {
```

Accessing

```cpp
m.member->DoThing()
```